### PR TITLE
Unlock background layer: correct history states and layer replacement

### PR DESCRIPTION
**Pull request by mcilroyc**
This PR provides proper background layer unlocking wrt to both history store, *and* the "replace layer" logic.

1. The replace layer logic now handles the special case of unlock background layer by allowing an explicit layer ID parameter of that which is to be replaced.  The recent simplifications to replace layers based on selection state does not handle this situation gracefully otherwise.
2. The unlock background layer action no longer emits two separate history-changing flux events

addresses #1443

### CODE REVIEW

File: src/js/actions/layers.js [hunk 1]
```diff
@@ -892,7 +894,7 @@ define(function (require, exports) {
             .bind(this)
             .then(function (event) {
                 var layerID = event.layerID;
-                return this.transfer(addLayers, document, layerID, true, true);
+                return this.transfer(addLayers, document, layerID, true, layer.id);
```
Comment (volfied) [inline src/js/actions/layers.js]: addLayersCommand's JSDoc suggests that replace takes an index, yet you're passing the layer ID here.

File: src/js/models/layerstructure.js [hunk 2]
```diff
@@ -713,28 +713,31 @@ define(function (require, exports, module) {
      * @param {Array.<object>} descriptors Photoshop layer descriptors
      * @param {boolean} selected Whether the new layer should be selected. If
      *  so, the existing selection is cleared.
-     * @param {boolean} replace Whether to replace the first existing layer model at
-     *  the given index
+     * @param {boolean= || number=} replace can be explicitly false, undefined, or a layer ID
```
Comment (volfied) [inline src/js/models/layerstructure.js]: I see the correct JSDoc here.

**Comment by volfied** [comment]
Got one JSDoc nit, otherwise changes look good to me @mcilroyc

**Comment by mcilroyc** [comment]
JSDoc fixed.  nice catch.  back to you @volfied

**Pull request merged**